docusaurus/docs/tutorials/build-a-streaming-data-source-plugin.md

```diff
@@ -15,143 +14,321 @@ keywords:
-   :::
+Now we need to add the necessary code to the backend. For that, we change `pkg/plugin/datasource.go`, this is where the backend part responsible for handling the queries created by the frontend is defined. In our case, as we want to handle a stream, we need to implement the `backend.StreamHandler`.
 
-### Example code for final `query` method
+1. Add the following part in the replacement of the `var`:
 
-```ts
-query(options: DataQueryRequest<MyQuery>): Observable<DataQueryResponse> {
-  const streams = options.targets.map(target => {
-    const query = defaults(target, defaultQuery);
+```tsx
+var (
+    _ backend.CheckHealthHandler    = (*Datasource)(nil)
+    _ instancemgmt.InstanceDisposer = (*Datasource)(nil)
+    _ backend.StreamHandler         = (*Datasource)(nil)
+)
+```
 
-    return new Observable<DataQueryResponse>(subscriber => {
-      const frame = new CircularDataFrame({
-        append: 'tail',
-        capacity: 1000,
-      });
+That means our `DataSource` will implement `backend.CheckHealthHandler`, `instancemgmt.InstanceDisposer`, and `backend.StreamHandler`. The methods necessary for the first two are already provided in the scaffold; therefore, we need to implement only the methods necessary for `backend.StreamHandler`.
 
-      frame.refId = query.refId;
-      frame.addField({ name: 'time', type: FieldType.time });
-      frame.addField({ name: 'value', type: FieldType.number });
+2. This code will be called when the user tries to subscribe to a channel. You can implement permissions checking here, but in our case we just want the user to connect successfully in each attempt; therefore, we just return a backend: `SubscribeStreamStatusOK`.
 
-      const intervalId = setInterval(() => {
-        frame.add({ time: Date.now(), value: Math.random() });
+3. Implement the `PublishStream` method:
 
-        subscriber.next({
-          data: [frame],
-          key: query.refId,
-          state: LoadingState.Streaming,
-        });
-      }, 100);
+```tsx
+func (d *Datasource) PublishStream(context.Context, *backend.PublishStreamRequest) (*backend.PublishStreamResponse, error) {
+    return &backend.PublishStreamResponse{
+        Status: backend.PublishStreamStatusPermissionDenied,
+    }, nil
+}
+```
 
-      return () => {
-        clearInterval(intervalId);
-      };
-    });
-  });
+This code is called whenever a user tries to publish to a channel. As we don't want to receive any data from the frontend after the first connection is created, we will just return a `backend.PublishStreamStatusPermissionDenied`.
 
-  return merge(...streams);
+4. To prevent a random error, change the `CheckHealth` method provided by default for the following code:
+
+```tsx
+func (d *Datasource) SubscribeStream(context.Context, *backend.SubscribeStreamRequest) (*backend.SubscribeStreamResponse, error) {
+    return &backend.SubscribeStreamResponse{
+        Status: backend.SubscribeStreamStatusOK,
+    }, nil
+}
+```
```

Comment:
I think this part here is not correct. We are saying that the `CheckHealth` should be replaced, but then we are showing the code for `SubscribeStream`. I think the correct core should be the `CheckHealth`

Reply:
fixed.